Source/WebCore/rendering/RenderReplaced.cpp

```diff
@@ -425,10 +427,17 @@ void RenderReplaced::computeAspectRatioInformationForRenderBox(RenderBox* conten
     // function was added, since all it has done is make the code more unclear.
     constrainedSize = intrinsicSize;
     if (intrinsicRatio && !intrinsicSize.isEmpty() && style().logicalWidth().isAuto() && style().logicalHeight().isAuto()) {
-        // We can't multiply or divide by 'intrinsicRatio' here, it breaks tests, like fast/images/zoomed-img-size.html, which
-        // can only be fixed once subpixel precision is available for things like intrinsicWidth/Height - which include zoom!
-        constrainedSize.setWidth(RenderBox::computeReplacedLogicalHeight() * intrinsicSize.width() / intrinsicSize.height());
-        constrainedSize.setHeight(RenderBox::computeReplacedLogicalWidth() * intrinsicSize.height() / intrinsicSize.width());
+        if (is<RenderVideo>(this)) {
```

Comment:
Sorry, with the changes I'm not sure about this one anymore. Do you think this is still an issue with the changes?

`computeIntrinsicRatioInformation` should now should compute `intrinsicRatio` correctly and not allow the default object size to influence it. If the object is a `RenderVideo`, then we can do what you said and use `intrinsicRatio`

Reply:
Yeah, looks fine now. However, I think here it would be more accurate to use `hasDefaultObjectSize` same to `computeIntrinsicRatioInformation`. Maybe we could move `hasDefaultObjectSize` to RenderVideo?

The other part looks good to me:)

Reply:
Ok yeah that would definitely make more sense. Added hasDefaultObjectSize to RenderVideo. I also modified the casting to RenderVideo to use dynamicCast so that it is more concise. How does it look now?

Reply:
I guess it should be `if (is<RenderVideo>(*this) && downcast<RenderVideo>(*this).hasDefaultObjectSize())` here?

Reply:
Hmmm wouldn't we want to have the code take this new path if we want to compute the video's constrained size using the aspect ratio? We would compute the width and then use the aspect ratio with it to compute the height. Does that sound right?

Reply:
Hmm, I think only video using the default size takes this path. Regarding other video, is there specification supported this change?

Reply:
Yeah I looked over it again and it looks like that change is fine and I went ahead and made it. What I guess I'm not sure about is how the code below in the else branch (the original code) is correct. I think I may be missing some sort of context, but it doesn't seem right to be computing both the width and height like that using the aspect ratio. That may be a conversation to have at another time though.....

Let me know what you think of the changes